.github/workflows/release.yml

```diff
@@ -0,0 +1,113 @@
+name: release
+
+on:
+  workflow_dispatch:
+    inputs:
+      publishPreRelease:
+        description: "Publish a pre-release ?"
+        required: true
+        type: choice
+        options:
+          - "true"
+          - "false"
+        default: "true"
+      publishToMarketPlace:
+        description: "Publish to VS Code Marketplace ?"
+        required: true
+        type: choice
+        options:
+          - "true"
+          - "false"
+        default: "false"
+      publishToOVSX:
+        description: "Publish to OpenVSX Registry ?"
+        required: true
+        type: choice
+        options:
+          - "true"
+          - "false"
+        default: "false"
+jobs:
+  packaging-job:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout vscode-knative
+        uses: actions/checkout@v4
+      - name: Use Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18
+      - name: Install dependencies
+        run: |
+          npm install -g typescript "vsce" "ovsx"
+          npm install
+          echo "EXT_VERSION=$(cat package.json | jq -r .version)" >> $GITHUB_ENV
+      - name: Build And Run Unit Tests
+        uses: coactions/setup-xvfb@b6b4fcfb9f5a895edadc3bc76318fae0ac17c8b3 #v1.0.1
+        with:
+          run: npm run test
+      - name: Package
+        run: |
+          jq --tab '.extensionDependencies += [ "ms-kubernetes-tools.vscode-kubernetes-tools" ]' package.json  > package.json.new
+          mv package.json.new package.json
+          node ./out/build/update-readme.js
+          declare -A targets
+          targets["win32-x64"]=win32
+          targets["win32-arm64"]=win32
+          targets["linux-x64"]=linux
+          targets["linux-arm64"]=linux-arm64
+          targets["darwin-x64"]=darwin
+          targets["darwin-arm64"]=darwin
+          for target in ${!targets[@]}; do
+            export TARGET=${targets[${target}]}
+            vsce package --target ${target} -o vscode-knative-${{ env.EXT_VERSION }}-${GITHUB_RUN_NUMBER}-${target}.vsix
+            sha256sum *-${target}.vsix > vscode-knative-${{ env.EXT_VERSION }}-${GITHUB_RUN_NUMBER}-${target}.vsix.sha256
+          done
```

Comment:
I don't think we need to package a vsix artifact per platform. vscode-knative is platform independent (see the existing releases). I would just have one package, and checksum validation and there's no need for anything related to `${target}`.

Same thing goes for the release logic, no need for `platform`. We can release just one universal vsix.